### PR TITLE
Optionally detect caret movement with events in terminal applications

### PR DESCRIPTION
**Pull request by codeofdusk**
### Link to issue number:
None

### Summary of the issue:
Using events to detect caret movement increases responsiveness in some scenarios, but we intermitantly get unreliable results when backspacing quickly (see microsoft/terminal#10820).

### Description of how this pull request fixes the issue:
Add a feature flag to enable `caretMovementDetectionUsesEvents` in terminals. The setting is currently ternary to avoid the need for a config spec upgrade should automatic enablement later be added in some apps.

In a follow-up PR, the checkbox may be changed to a combobox (similar to the approach in #10964).

### Testing strategy:
Tested that caret events are used when the feature flag is enabled, and not used when the flag is disabled in Windows Terminal and both UIA and legacy conhost.

### Known issues with pull request:
None.

### Change log entries:
== Changes for Developers ==
- Added an advanced setting to enable caret movement detection in terminal applications for testing purposes.

### Code Review Checklist:

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual testing.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers

### CODE REVIEW

**Comment by codeofdusk** [comment]
Cc @DHowett, @carlos-zamora, @miniksa.

File: source/NVDAObjects/UIA/winConsoleUIA.py [hunk 1]
```diff
@@ -410,4 +417,9 @@ def findExtraOverlayClasses(obj, clsList):
 
 
 class WinTerminalUIA(EnhancedTermTypedCharSupport):
-	pass
+	def _get_caretMovementDetectionUsesEvents(self):
```
Comment (leonardder) [inline source/NVDAObjects/UIA/winConsoleUIA.py:420]: Isn't this the default? Also, I don't think this needs to be a property.

**Comment by codeofdusk** [comment]
> Isn't this the default?

[Nope](https://github.com/nvaccess/nvda/blob/master/source/NVDAObjects/behaviors.py#L361).

File: source/NVDAObjects/UIA/winConsoleUIA.py (hunk 1 above)
Comment (codeofdusk) [reply source/NVDAObjects/UIA/winConsoleUIA.py]: > Isn't this the default?

[Nope](https://github.com/nvaccess/nvda/blob/master/source/NVDAObjects/behaviors.py#L361).

**Review by seanbudd — DISMISSED** [review]

**Comment by feerrenrut** [comment]
At this late stage there needs to be strong justification for why the 2021.3 release can't be made without this change, otherwise it should target 2022.1

**Comment by codeofdusk** [comment]
PR retargetted to master (2022.1).

**Comment by codeofdusk** [comment]
I've rebased/retargetted the PR for 2022.1. Cc @seanbudd and I think @Qchristensen?

**Review by feerrenrut — CHANGES_REQUESTED** [review]

**Pull request closed without merging**